llvm/include/llvm/MC/MCSymbolRepo.h

```diff
@@ -23,6 +23,19 @@ class MCSymbolRepo : public MCSymbol {
 
   // A pointer to the TiketNode metadata of the corresponding GlobalObject.
   const TicketNode *CorrespondingTicketNode = nullptr;
+
+  // Get the symbol full name.
+  static std::string getFullName(const MCContext &Ctx,
+                                 const StringRef InitialName,
+                                 const ticketmd::DigestType &Digest) {
+    std::string Name =
+        Ctx.getAsmInfo()->getPrivateGlobalPrefix().str() + InitialName.str();
```

Comment:
Because there is a bug in the previous version. Currently, the `PrivateGlobalPrefix` for Reop is set to an empty string, which hides the issue.

For example: 
```
MCSection *TargetLoweringObjectFileRepo::getExplicitSectionGlobal(
    const GlobalObject *GO, SectionKind Kind, const TargetMachine &TM) const {
  return selectRepoSectionForGlobal(getContext(), GO, Kind,
                                    cast<MCSymbolRepo>(TM.getSymbol(GO)));
}

MCSection *TargetLoweringObjectFileRepo::SelectSectionForGlobal(
    const GlobalObject *GO, SectionKind Kind, const TargetMachine &TM) const {
  return selectRepoSectionForGlobal(getContext(), GO, Kind,
                                    cast<MCSymbolRepo>(TM.getSymbol(GO)));
}
```
We could see in the function of TM.getSymbol(GO), it use the `getNameWithPrefix`.
In the repo object writer, we also write the private global symbol with a prefix, please see the RepoObjectWriter::getSymbolName (previous version) for the detail. 

Since the private global symbols are saved in the datatbase with a `PrivateGlobalPrefix` string, the pruning pass should use the same name. Therefore, I added a function here, which is used for both pruning and object writer.

Reply:
Yes, that makes sense.